BaseTools/Plugin/HostBasedUnitTestRunner/HostBasedUnitTestRunner.py

```diff
@@ -180,14 +192,22 @@ def gen_code_coverage_gcc(self, thebuilder):
         regex_exclude = r"^.*UnitTest\|^.*MU\|^.*Mock\|^.*DEBUG"
         # MU_CHANGE end - regex string for exclude paths
 
+        lcov_version_major = self.get_lcov_version()
+        if not lcov_version_major:
+            logging.error("UnitTest Coverage: Failed to determine lcov version")
+            return 1
+        logging.info(f"Got lcov version {lcov_version_major}")
+
         # Generate base code coverage for all source files
         ret = RunCmd("lcov", f"--no-external --capture --initial --directory {buildOutputBase} --output-file {buildOutputBase}/cov-base.info --rc lcov_branch_coverage=1")
         if ret != 0:
             logging.error("UnitTest Coverage: Failed to build initial coverage data.")
             return 1
 
         # Coverage data for tested files only
-        ret = RunCmd("lcov", f"--capture --directory {buildOutputBase}/ --output-file {buildOutputBase}/coverage-test.info --rc lcov_branch_coverage=1")
+        # `--ignore-errors mismatch` needed to make lcov v2.0+/gcov work.
+        lcov_error_settings = "--ignore-errors mismatch" if lcov_version_major >= 2 else ""

```

Comment:
Is there any reason _not_ to just always have this flag enabled? It would reduce this complexity, not that there is much.

Reply:
Ah. I see this came from edk2.

Reply:
I made a similar change, eventually search led to the lcov issue and then saw the edk2 change linked there.

Reply:
Will keep as-is to minimize non-essential delta.